pylint_plugins/bundled_imports_checker.py

```diff
@@ -0,0 +1,229 @@
+
+    def leave_module(self, node):
+        # type: (scoped_nodes.Module) -> None
+        """
+        We perform the actual check just before we leave the module since that's when the file has
+        been fully processed.
+        """
+        node_file_path = self._get_file_path_for_node(node=node)
+
+        third_party_nodes = list(
+            self._third_party_import_nodes.get(node_file_path, [])
+        )  # type: List[node_classes.Import]
+        scalyr_init_node = self._scalyr_init_nodes.get(node_file_path, None)
+
+        for import_node in third_party_nodes:
+            self._verify_import_node_comes_after_scalyr_init(
+                scalyr_init_node=scalyr_init_node, import_node=import_node
+            )
+
+    def _verify_import_node_comes_after_scalyr_init(
+        self, scalyr_init_node, import_node
+    ):
+        """
+        Verify any import which references a bundled third party module comes after scalyr_init()
+        call.
+        """
+        if not scalyr_init_node:
+            # TODO: Implement support for indirect imports. Currently we only check files which
+            # contain "scalyr_init()" call aka main entry points.
+            return
+            args = (import_node.name, import_node.lineno)
+            self.add_message(
+                self.SCALYR_INIT_CALL_NOT_FOUND, node=import_node, args=args
+            )
+
+        if import_node.lineno < scalyr_init_node.lineno:
+            args = (import_node.name, import_node.lineno, scalyr_init_node.lineno)
+            self.add_message(
+                self.INVALID_BUNDLED_IMPORT_ORDER, node=import_node, args=args
+            )
+            return
+
+    def _get_file_path_for_node(self, node):
+        # Recursievly walk up to the top most parent to find the file original node comes from
+        if getattr(node, "file", None):
+            return node.file
+        elif node.parent:
+            return self._get_file_path_for_node(node=node.parent)
+
+        return None
+
+
+def get_third_party_package_module_names():
+    # type: () -> List[str]
+    """
+    Return a list of bundled 3rd party package and module names.
+    """
+    result = []  # type: List[str]
+
+    def is_python_package(directory_path, file_path):
+        # type: (str, str) -> Tuple[bool, Optional[str]]
+        """
+        Return package name if the provided file path is a Python package, None otherwise.
+        """
+        file_name = os.path.basename(file_path)
+        init_file_path = os.path.join(file_path, "__init__.py")
+
+        if os.path.isdir(file_path) and os.path.isfile(init_file_path):
+            # Package
+            return (True, file_name)
+
+        return (False, None)
+
+    def is_python_module(directory_path, file_path):
+        # type: (str, str) -> Tuple[bool, Optional[str]]
+        """
+        Return package name if the provided file path is a Python package, None otherwise.
```

Comment:
The comment is the same as the method above.

Reply:
Pushed a fix - d0ae301cdd0fed2b9324a9e98ea3853808e53b3a.